src/core/varinfo.jl

```diff
@@ -153,64 +153,32 @@ syms(vi::VarInfo) = map(vn -> vn.sym, vns(vi))  # get all symbols
 
 # The default getindex & setindex!() for get & set values
 # NOTE: vi[vn] will always transform the variable to its original space and Julia type
-Base.getindex(vi::VarInfo, vn::VarName) = begin
-  @assert haskey(vi, vn) "[Turing] attempted to replay unexisting variables in VarInfo"
-  dist = getdist(vi, vn)
-  # if isa(dist, SimplexDistribution) || isa(dist, MvNormal) # Reduce memory allocation for distributions with simplex constraints
-  #   if vn in keys(vi.rvs)
-  #     r = vi.rvs[vn]
-  #     reconstruct!(r, dist, getval(vi, vn))
-  #   else
-  #     r = reconstruct(dist, getval(vi, vn))
-  #     r_real = similar(r, Real)
-  #     r_real[:] = r
-  #     vi.rvs[vn] = r_real
-  #     r = r_real
-  #   end
-  #   istrans(vi, vn) ?
-  #     invlink!(r, dist, r) :
-  #     r
-  # else
-    istrans(vi, vn) ?
-      invlink(dist, reconstruct(dist, getval(vi, vn))) :
-      reconstruct(dist, getval(vi, vn))
-  # end
+function Base.getindex(vi::VarInfo, vn::VarName)
+    @assert haskey(vi, vn) "[Turing] attempted to replay unexisting variables in VarInfo"
+    dist = getdist(vi, vn)
+    return copy(istrans(vi, vn) ?
+        invlink(dist, reconstruct(dist, getval(vi, vn))) :
+        reconstruct(dist, getval(vi, vn)))
 end
 
 Base.setindex!(vi::VarInfo, val::Any, vn::VarName) = setval!(vi, val, vn)
 
-Base.getindex(vi::VarInfo, vns::Vector{VarName}) = begin
-  @assert haskey(vi, vns[1]) "[Turing] attempted to replay unexisting variables in VarInfo"
-  dist = getdist(vi, vns[1])
-  # if isa(dist, SimplexDistribution) # Reduce memory allocation for distributions with simplex constraints
-  #   if vns in keys(vi.rvs)
-  #     r = vi.rvs[vns]
-  #     reconstruct!(r, dist, getval(vi, vn))
-  #   else
-  #     r = reconstruct(dist, getval(vi, vns), length(vns))
-  #     r_real = similar(r, Real)
-  #     r_real[:] = r
-  #     vi.rvs[vns] = r_real
-  #     r = r_real
-  #   end
-  #   istrans(vi, vns[1]) ?
-  #     invlink!(r, dist, r) :
-  #     r
-  # else
-    istrans(vi, vns[1]) ?
-      invlink(dist, reconstruct(dist, getval(vi, vns), length(vns))) :
-      reconstruct(dist, getval(vi, vns), length(vns))
-  # end
+function Base.getindex(vi::VarInfo, vns::Vector{VarName})
+    @assert haskey(vi, vns[1]) "[Turing] attempted to replay unexisting variables in VarInfo"
+    dist = getdist(vi, vns[1])
+    return copy(istrans(vi, vns[1]) ?
```

Comment:
This looks a bit strange. Do you want to:

- `copy(istrans(vi, vns[1])` or
- `copy(invlink(dist, reconstruct(dist, getval(vi, vns), length(vns))))` and `copy( reconstruct(dist, getval(vi, vns), length(vns))))`

Reply:
I see. Might be helpful to introduce an additional variable here for readability 

```julia

res = istrans(vi, vns[1]) ?
        invlink(dist, reconstruct(dist, getval(vi, vns), length(vns))) :
        reconstruct(dist, getval(vi, vns), length(vns))
return res
```